puppet/zulip/files/hooks/pre-deploy.d/zulip_notify.hook

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+# Arguments: OLD_COMMIT NEW_COMMIT ...where both are `git describe`
+# output or tag names.  The CWD will be the new deploy directory.
+
+set -e
+set -u
+
+if ! ZULIP_API_KEY=$(crudini --get /etc/zulip/zulip-secrets.conf secrets zulip_release_api_key); then
+    echo "zulip_notify: No zulip_release_api_key set!  Set zulip_release_api_key in /etc/zulip/zulip-secrets.conf"
+    exit 0
+fi
+
+if ! ZULIP_NOTIFY_BOT_EMAIL=$(crudini --get /etc/zulip/zulip.conf zulip_notify bot_email); then
+    echo "zulip_notify: No zulip_notify.bot_email set in /etc/zulip/zulip.conf"
+    exit 0
+fi
+
+if ! ZULIP_NOTIFY_SERVER=$(crudini --get /etc/zulip/zulip.conf zulip_notify server); then
+    echo "zulip_notify: No zulip_notify.server set in /etc/zulip/zulip.conf"
+    exit 0
+fi
```

Comment:
Hmm, we could save some of this parsing if we had the parameter be the path to a `zuliprc` file for credentials rather than passing so many parameters. But I guess we'd still need the parameter for what stream to post in.

Reply:
Yeah, I thought about that.  It's one additional file to sync between prod and staging, and possibly not as clear of an error.